controller/backup_controller.go

```diff
@@ -413,153 +415,122 @@ func (bc *BackupController) validateBackingImageChecksum(volName, biName string)
 	return bi.Status.Checksum, nil
 }
 
-func (bc *BackupController) backupCreation(backupTarget *longhorn.BackupTarget, volume *longhorn.Volume, backup *longhorn.Backup) (skip bool, err error) {
-	log := bc.logger.WithFields(
-		logrus.Fields{
-			"backup":   backup.Name,
-			"vol":      volume.Name,
-			"snapshot": backup.Spec.SnapshotName,
-			"label":    backup.Spec.Labels,
-		},
-	)
-
-	currentState := backup.Status.State
+// backupCreation calls engine's snapshot backup API or sync current engine's snapshot backup progress/state
+// if the `spec.SnapshotName` is set and `status.lastSyncedAt` is unset.
+// note that:
+// - if the CR without the spec.SnapshotName, the sourcce of truth is the remote backup target
+// - if the CR with the spec.SnapshotName, there are two source of truth:
+//   - the 1st is the engine, longhorn manager calls engine's snapshot backup API and monitoring the longhorn engine reports progress/state
+//   - the 2nd is once the engine reports snapshot backup state Completed, the source of truth becomes the remote backup target
```

Comment:
1. Move these comments above `if backup.Status.LastSyncedAt.IsZero() && backup.Spec.SnapshotName != "" {`.
2. Field `spec.SnapshotName` should be set once the CR is created. How can it become empty? Do you mean the source of truth is the engine/replica if `status.lastSyncedAt` is empty?
3. If `status.lastSyncedAt` is set, it means the controller won't sync with the monitor hence the source of truth should be the remote backup target only rather than both.

Reply:
If the Backup CR is pulled from the remote backup target, the `spec.SnapshotName` would be empty.

I think the source of truth is:
- engine/replica is `status.lastSyncedAt == ""` && `spec.SnapshotName != ""` && `status.State != Complete`.
- remote backup target is `status.lastSyncedAt == ""` && `spec.SnapshotName != ""` && `status.State == Complete` _or_ `status.lastSyncedAt == ""` && `spec.SnapshotName == ""`.

Reply:
OK. Got it. If the backup CR is created by the backup volume controller and needs to be synced with the remote backup target immediately after creation, the snapshot name is empty.

The condition for the remote backup target is too complicated. As long as the source of truth is not the engine/replica, then it must be the remote backupstore.

Reply:
Okay, I'll make the wording/condition be simplified.

Reply:
Updated!